### PR TITLE
feat: 1829 flex invalid prior notice day range

### LINKED ISSUE #1829: Flex: invalid_prior_notice_day_range
### Describe the problem

See below

### Describe the new validation rule

[`invalid_prior_notice_day_range` as defined in the table here](https://docs.google.com/document/d/1QABoIeOdR6lAB7A6_X6WIbZwC7EnAIZvm_hRNwNQeoI/edit#heading=h.fhjv4kui4obw)

### Sample GTFS datasets

_No response_

### Severity

Error

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qcdyx**
**Summary:**
Closes #1829 
add a new notice for flex feed: invalid_prior_notice_day_range

**Expected behavior:** 
if the flex feed's prior last day is greater than prior start day, then generate this error.
image

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ ] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Comment by emmambd** [comment]
@qcdyx On a QA side, works as expected! Here's the test data I made for this. 

[dsi.zip](https://github.com/user-attachments/files/17214444/dsi.zip)

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/BookingRulesEntityValidator.java [hunk 1]
```diff
@@ -144,4 +153,34 @@ static class ForbiddenSameDayBookingFieldValueNotice extends ValidationNotice {
       this.fieldNames = String.join(", ", forbiddenFields);
     }
   }
+
+  /**
```
Comment (jcpitre) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/BookingRulesEntityValidator.java:157]: Is this the info that you want highlighted in rules.html? 
Usually there is a short description that will be displayed in bold, followed by a longer explanation.  Your description seems a bit long to be the short description.

Comment (qcdyx) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/BookingRulesEntityValidator.java:157]: Ok, shortened it to   /** prior notice last day should not be greater than the start day in booking_rules.txt. */

**Comment by qcdyx** [comment]
@jcpitre 
image

**Review by jcpitre — APPROVED** [review]
LGTM

**Comment by emmambd** [comment]
I'm confused by the "prior notice last day earlier than start day" addition. Can we remove this?

**Pull request merged**